src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -70,14 +70,16 @@ public virtual IDisposable Start()
             var server = ServerFactory.Start(_serverInstance,
                 async features =>
                 {
-                    var httpContext = contextFactory.CreateHttpContext(features);
-                    httpContext.ApplicationServices = _applicationServices;
-                    var requestIdentifier = GetRequestIdentifier(httpContext);
-
-                    using (logger.BeginScope("Request Id: {RequestId}", requestIdentifier))
+                    using (var httpContext = contextFactory.CreateHttpContext(features))
```

Comment:
Removed ```using``` and added ```httpContext.Response.RegisterForDispose(httpContext);```

Will need to change it so Disposing is done after OnCompleted rather than during; but probably should be anyway?